### ISSUE TITLE
"Ignore outliers in chart scaling" made me think my early training data was lost

### ISSUE THREAD

**Issue by nealwu** [tensorflow/tensorboard#214]
I started retraining a model from scratch after changing its implementation, and I wanted to compare my results from the first few epochs between the two versions of the model. I went on Tensorboard for the original model and realized I couldn't see the accuracy for the first 3-4 epochs at all.

I ended up believing that the first few epochs had been lost somehow until @reedwm suggested unchecking "Ignore outliers in chart scaling." Once I did, I magically got all my data back.

I understand the rationale for having this feature is to zoom in on the more recent data which the user is probably more interested in, rather than having to zoom out to show the initial steep climb/drop. I wouldn't mind if it were simply a default zoom option, where you could still zoom out to see the rest of your data. However, if you aren't aware of this checkbox, your "outlier" data is completely inaccessible.

cc @chihuahua

**Comment by nealwu** [comment]
Here are two screenshots with the box checked and unchecked for reference. When it is checked, I can only see my accuracy from step 160,000 onward. After unchecking, I can see the accuracy at step 40,000.
![screenshot from 2017-07-10 16 26 54](https://user-images.githubusercontent.com/726075/28044545-ddf8d2d0-658c-11e7-88ab-3df1ebd28ccb.png)
![screenshot from 2017-07-10 16 27 03](https://user-images.githubusercontent.com/726075/28044547-de12ffc0-658c-11e7-9192-7dacf16deb42.png)

**Comment by dandelionmane** [comment]
Hmm, that is confusing.

I think most of the time, hiding outliers is helpful to users. But maybe the cost of occasionally not knowing where the data went outweighs the benefit of having the chart do a useful thing most of the time. If we were to adopt something like a principle of least surprise, we would default that option to off (because not showing all of the data is surprising) and then power users who need the feature can search for it.

@wchargin @chihuahua @jart Do you have thoughts on this?

**Comment by nealwu** [comment]
I would also be fine with defaulting to zoomed in, as long as there's a more obvious button next to the graph that enables you to zoom out and see all your data.

I tried both of the buttons next to the graph, which didn't solve my problem, and then I concluded that the data I was looking for was gone. What I really didn't expect was that it was a checkbox all the way to the left side.

**Comment by chihuahua** [comment]
I lack an ideal solution. Hiding outliers had been motivated by an email thread noting that the steep drop in the beginning is often not useful. However, it does seem reasonable to me that the current UI is confusing because users might not look to the far left in order to resolve their view of the chart. Here is another user who I think finds the toggle unclear: https://github.com/tensorflow/tensorboard/issues/273

I lean towards Dandelion's suggestion of making "Ignore outliers in chart scaling" off by default and letting power users turn it on. How about we do that?

This might be a case in which we can't fully satisfy every user, but between clarity and convenience, I prefer the former.

Sorry for letting this issue drop.

**Comment by wchargin** [comment]
> making "Ignore outliers in chart scaling" off by default and letting power users turn it on.

The problem that I have with this is the following. You almost always _do_ want to ignore outliers. If it's enabled by default, there is some potential for confusion exactly once, after which you've learned the UI and everything works. But if it's disabled by default, then (a) users who know that it exists will have to turn it on (and remember to turn it on) every time that they open TensorBoard, and (b) the same users who didn't notice it in the other case will not notice it in this case, and thus will _never_ get its benefits.

I suspect that if we set it to be off by default, we will have more unhappy users than if we leave it default-on.

**Comment by nealwu** [comment]
It's not as simple as "confusion exactly once." I would likely have been misled into forever thinking that my first few data points were gone, if not for the fact that I work on the Brain team and a co-worker specifically told me to uncheck that box. If instead I were working on a model on my own, I don't think I would have realized this at any point.

Having to turn the feature on every time you run TensorBoard is inconvenient, but you can't make it convenient for everyone. Personally I turn this feature off every time I open TensorBoard (I always want to see my whole graph) and also set smoothing to zero.

Here are a couple of middle-ground alternatives that could satisfy both concerns:
- Start with the graph zoomed in, but add a button that will zoom out and show you the whole graph. This button should be *next to the graph* so that it is obvious that I need to click on it to see more of my data.
- Set the graphs to log scale by default, so that the steep climb/drop at the start is less extreme.

**Comment by nealwu** [comment]
I personally favor the first approach of those two, since it doesn't make anything worse for users who like their graphs zoomed in, and at the same time it becomes much more obvious how to access the full graph when you need it.

**Comment by nealwu** [comment]
One potential solution which may be easier: keep things as is, but show the black box with the numerical values even for data points that are off the chart.
